internal/services/azurestackhci/stack_hci_storage_path_resource.go

```diff
@@ -0,0 +1,238 @@
+package azurestackhci
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/tags"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/azurestackhci/2024-01-01/storagecontainers"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/extendedlocation/2021-08-15/customlocations"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var (
+	_ sdk.Resource           = StackHCIStoragePathResource{}
+	_ sdk.ResourceWithUpdate = StackHCIStoragePathResource{}
+)
+
+type StackHCIStoragePathResource struct{}
+
+func (StackHCIStoragePathResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return storagecontainers.ValidateStorageContainerID
+}
+
+func (StackHCIStoragePathResource) ResourceType() string {
+	return "azurerm_stack_hci_storage_path"
+}
+
+func (StackHCIStoragePathResource) ModelObject() interface{} {
+	return &StackHCIStoragePathResourceModel{}
+}
+
+type StackHCIStoragePathResourceModel struct {
+	Name              string                 `tfschema:"name"`
+	ResourceGroupName string                 `tfschema:"resource_group_name"`
+	Location          string                 `tfschema:"location"`
+	CustomLocationId  string                 `tfschema:"custom_location_id"`
+	Path              string                 `tfschema:"path"`
+	Tags              map[string]interface{} `tfschema:"tags"`
+}
+
+func (StackHCIStoragePathResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
does the `name` have max length and special characters limitation?

Reply:
updated